### PR TITLE
Fix #1391 Improve QueryDSL's union query compatibility with PostgreSQL

### LINKED ISSUE #1391: QueryDSL doesn't correctly union SELECT...ORDER BY/LIMIT statements for PostgreSQL
When using `UNION`, with `ORDER BY` or `LIMIT` within a subexpression's `SELECT` statement, PostgreSQL requires the `SELECT` statement be enclosed in parentheses: https://stackoverflow.com/a/24024639 However, the following QueryDSL code causes a syntax error against Postgres because it doesn't generate those parentheses:

```scala
val i = Item.syntax("i")
selectFrom(Item.as(i)).orderBy(i.id).unionAll(
  selectFrom(Item.as(i)).orderBy(i.id)
).list().apply()
```

My current workaround is to use `.toSQLSyntax` on each subexpression and combine them manually with `sqls"($a) union all ($b)"`

### PR DESCRIPTION
**Pull request by keyno63**
fix #1391

- Add round bracket when using union syntax.
- A reason a function is branched is mysql seems to disallow double round bracket (even so postgres and h2 allow it.)

### CODE REVIEW

**Review by seratch — CHANGES_REQUESTED** [review]
Adding round brackets for subquery parts should be fine as the CI builds pass with all the supported databases.

File: scalikejdbc-interpolation/src/main/scala/scalikejdbc/QueryDSLFeature.scala
Comment (keyno63) [reply scalikejdbc-interpolation/src/main/scala/scalikejdbc/QueryDSLFeature.scala]: Thanks for reviewing.

I changed the function name.
6d2d16d409ce02158461e28f776c9b00cbf25df2

File: scalikejdbc-interpolation/src/main/scala/scalikejdbc/QueryDSLFeature.scala
Comment (keyno63) [reply scalikejdbc-interpolation/src/main/scala/scalikejdbc/QueryDSLFeature.scala]: I did not notice it.
I fixed to trimming spaces.
6d2d16d409ce02158461e28f776c9b00cbf25df2

**Review by seratch — APPROVED** [review]
Look good to me! We can merge this PR after waiting for others' feedback for a few more days.

**Comment by keyno63** [comment]
Thank you for reviewing this PR.

 I'm excited that my work may contribute to a project I love!

**Pull request merged**
